opentelemetry-sdk/benches/log.rs

```diff
@@ -0,0 +1,119 @@
+            .build();
+        let tracer = tracer_provider.tracer("bench-tracer");
+
+        // Act
+        tracer.in_span("bench-span", |_cx| {
+            b.iter(|| f(&logger));
+        });
+    });
+
+    group.finish();
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    log_benchmark_group(c, "simple-log", |logger| {
+        logger.emit(LogRecord::builder().with_body("simple log".into()).build())
+    });
+
+    log_benchmark_group(c, "long-log", |logger| {
+        logger.emit(LogRecord::builder().with_body("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Gravida in fermentum et sollicitudin ac orci phasellus. Ullamcorper dignissim cras tincidunt lobortis feugiat vivamus at augue. Magna etiam tempor orci eu. Sed tempus urna et pharetra pharetra massa.".into()).build())
+    });
+
+    let now = SystemTime::now();
+    log_benchmark_group(c, "full-log", |logger| {
+        logger.emit(
+            LogRecord::builder()
+                .with_body("full log".into())
+                .with_timestamp(now)
+                .with_observed_timestamp(now)
+                .with_severity_number(Severity::Warn)
+                .with_severity_text(Severity::Warn.name())
+                .build(),
+        )
+    });
+
+    log_benchmark_group(c, "full-log-with-4-attributes", |logger| {
+        logger.emit(
+            LogRecord::builder()
+                .with_body("full log".into())
+                .with_timestamp(now)
+                .with_observed_timestamp(now)
+                .with_severity_number(Severity::Warn)
+                .with_severity_text(Severity::Warn.name())
+                .with_attribute("name", "my-event-name")
+                .with_attribute("event.id", 20)
+                .with_attribute("user.name", "otel")
+                .with_attribute("user.email", "otel@opentelemetry.io")
+                .build(),
+        )
+    });
+
+    log_benchmark_group(c, "full-log-with-9-attributes", |logger| {
+        logger.emit(
+            LogRecord::builder()
+                .with_body("full log".into())
+                .with_timestamp(now)
+                .with_observed_timestamp(now)
+                .with_severity_number(Severity::Warn)
+                .with_severity_text(Severity::Warn.name())
+                .with_attribute("name", "my-event-name")
+                .with_attribute("event.id", 20)
+                .with_attribute("user.name", "otel")
+                .with_attribute("user.email", "otel@opentelemetry.io")
+                .with_attribute("log.source.file.name", "log.rs")
+                .with_attribute("log.source.file.path", "opentelemetry_sdk/benches/log.rs")
+                .with_attribute("log.source.file.line", 96)
+                .with_attribute("log.module.path", "opentelemetry_sdk::benches::log")
+                .with_attribute("log.target", "opentelemetry_sdk::benches::log")
```

Comment:
Good to have the test for all the attribute types (or at-least - bool, int, float, string, byte) - https://github.com/open-telemetry/opentelemetry-rust/blob/443f68b1b8cd9016aec7f21ce913884f473c848c/opentelemetry/src/logs/record.rs#L79-L95)

Reply:
Ok. I can have a test for each attributes type